### ISSUE TITLE
Tasks permission keeps failing after a week

### ISSUE THREAD

**Issue by Kalytis** [andstatus/todoagenda#108]
Similar to #101 but not fixed at all.

It seems like permission to access the tasks cancels itself after a while (approx. one week), causing tasks to disappear from the widget.
Reinstalling fixs the problem for another week. Setting permission to "Deny" doesn't cause the app to prompt again. Downgrading the app doesn't fix.

OnePlus 7T Pro

**Comment by yvolk** [comment]
This is Android system that should remember these permissions. So I would say that that this is your device firmware bug.
Just a guess: maybe the bug shows when your tasks app is being updated.

**Comment by Kalytis** [comment]
I'm using https://github.com/tasks/tasks, which is supposed to be recommended for Etar. And permissions removal isnt correlated to app updates.

**Comment by yvolk** [comment]
In any case our app doesn't even try to store permissions anywhere.
You could lookup in your device setting a list of permissions currently granted to any app.
So I guess after the next permissions problem appearance you will see there that some permission was lost?!

**Comment by Kalytis** [comment]
![image](https://user-images.githubusercontent.com/31107370/185784367-67589b8e-4036-473c-95ab-4eed613ca491.png)
![Screenshot_20220821-111610](https://user-images.githubusercontent.com/31107370/185784381-d173aa59-4574-4737-a388-014926ca733e.jpg)

Okay, by "luck" the problem just occurred again, so here's what I see. Sorry for the French.

My tasks disappear from the widget. Then when I go in the Calendars section of the settings, I can see that under the tasks list (here "Personnal" and "University"), it says "Not found". Then, when I go to the app settings, it says that the permissions are still granted to Todo...

I'm puzzled, I have to say.

**Comment by yvolk** [comment]
"Not found" in this list means that our widget does remember that some tasks list with such name was previously selected for showing in the widget. But currently this "tasks list" is not found among the whole list of available "tasks lists" provided by the "Tasks app" installed on your device.
This may mean that currently ToDo Agenda" app is not allowed (or fails for other reason) to retrieve the list...
Maybe the problem lays on the Tasks app side: its decrstion of "Android intents" to query data from it...

**Comment by Kalytis** [comment]
Then I'm quite unsure what my next course of action could be.

**Comment by yvolk** [comment]
@Kalytis Permissions for other apps (including for ToDo Agenda) to access tasks lists are controlled by the Tasks app that you are using.
So I think that you need to raise an 8ssue in that Tasks app issues. Referring to this discussion, of course.

**Comment by Kalytis** [comment]
Hi, by any chance have you got a new idea about that, please ?

**Comment by yvolk** [comment]
Hi @Kalytis.
I made a test on Android 13:
1. Installed Tasks app, latest version from F-Droid (v.13...) And added a couple of tasks
2. Installed ToDo Agenda, latest version. Created a Widget, granted permissions and see the two tasks in the widget.
 
![image](https://user-images.githubusercontent.com/207716/208099784-47f1239f-36aa-4ae0-ad3e-eba0b338511b.png)

3. I navigated to Settings -> Permission Manager -> Tasks Permission and Disallowed the permission for ToDo Agenda:
![image](https://user-images.githubusercontent.com/207716/208100213-e26e7d97-9283-4738-b85a-bb8d74509f36.png)

4. I returned to the Home screen of the Device and I immediately see a prompt to grant permissions
![image](https://user-images.githubusercontent.com/207716/208100463-5c9ed803-ab92-48c9-9e56-53dc67318118.png)

5. Grant permission by clicking on the widget, and after returning to Home screen I do see the same tasks again...
- so this is really not the same situation as you observe.

My advice is in a case of a problem, in order not to re-install the app to try this:
1. Remove Tasks permission for ToDo Agenda, and then explicitly grant it again. 
2. Restart your device after re-granting Tasks permission in a case nothing appeared.
